### PR TITLE
gBoards Ginny

### PR DESCRIPTION
**Pull request by germ**
Part two of the gboards-consolidation project. The Mega PR has been separated out into smaller per-keyboard PRs.

This PR contains a single keyboard as per @zvecr , in this case FaunchPad. This code was pulled from #8262 

- [x] Enhancement/optimization
- [x] Keymap/layout/userspace (addition or update)

## Issues Fixed or Closed by This PR

- Replaces #8262
- Waiting on #8921

### CODE REVIEW

File: keyboards/gboards/k/ginny/matrix.c [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+static matrix_row_t read_cols(uint8_t row) {
+    if (row < MATRIX_I2C) {
+        if (mcp23018_status) {  // if there was an error
+            return 0;
+        } else {
+            uint8_t data    = 0;
+            mcp23018_status = i2c_start(I2C_ADDR_WRITE, I2C_TIMEOUT);
+            if (mcp23018_status) goto out;
+            mcp23018_status = i2c_write(GPIOB, I2C_TIMEOUT);
+            if (mcp23018_status) goto out;
```
Comment (drashna) [inline keyboards/gboards/k/ginny/matrix.c:222]:
```suggestion
```

File: keyboards/gboards/ginny/readme.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+* Hardware Supported: Ginny (Kit, Partial, Ready)   
+* Hardware Availability: [gboards.ca](http://gboards.ca)  
+
+---
+
+## Firmware Building
+
+Build and flash with the command below. Be sure to reset your keyboard!
+
+    make gboards/ginny:default:dfu
```
Comment (fauxpark) [inline keyboards/gboards/ginny/readme.md:29]:
```suggestion
    make gboards/ginny:default:flash
```

File: keyboards/gboards/ginny/rules.mk [hunk 3]
```diff
@@ -0,0 +1,20 @@
+# MCU name
+MCU                 =   atmega32u4
```
Comment (fauxpark) [inline keyboards/gboards/ginny/rules.mk:2]: I would prefer this file follow the template more closely.

**Comment by kitchen** [comment]
any update on this? I just got a ginny and am trying to figure out how to program it for AOEIHTNS usage :)

File: keyboards/gboards/ginny/matrix.c [hunk 4]
```diff
@@ -0,0 +1,281 @@
...
+    return true;
+}
+
+inline bool         matrix_is_on(uint8_t row, uint8_t col) { return (matrix[row] & ((matrix_row_t)1 << col)); }
+inline matrix_row_t matrix_get_row(uint8_t row) { return matrix[row]; }
+
+void matrix_print(void) {
+    print("\nr/c 0123456789ABCDEF\n");
+    for (uint8_t row = 0; row < MATRIX_ROWS; row++) {
+        phex(row);
```
Comment (hunner) [inline keyboards/gboards/ginny/matrix.c:192]:
```suggestion
        print_hex8(row);
```

File: keyboards/gboards/ginny/matrix.c [hunk 5]
```diff
@@ -0,0 +1,281 @@
...
+        print(": ");
+        pbin_reverse16(matrix_get_row(row));
```
Comment (hunner) [inline keyboards/gboards/ginny/matrix.c:194]:
```suggestion
        print_bin_reverse16(matrix_get_row(row));
```

File: keyboards/gboards/ginny/rules.mk [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+BOOTLOADER          =   atmel-dfu
+
+CUSTOM_MATRIX       =   yes
+VIRTSER_ENABLE      =   no
+NKRO_ENABLE         =   no
+DEBOUNCE_TYPE       =   eager_pr
```
Comment (hunner) [inline keyboards/gboards/ginny/rules.mk:8]:
```suggestion
DEBOUNCE_TYPE       =   sym_eager_pr
```

**Comment by keyboard-magpie** [comment]
Germ sadly seems to have left the keyboard world for now, I think closing this for now until some other community member has the energy to take this up again is best.

**Pull request closed without merging**